swiftide-integrations/src/lancedb/retrieve.rs

```diff
@@ -124,6 +127,69 @@ impl Retrieve<SimilaritySingleEmbedding> for LanceDB {
     }
 }
 
+#[async_trait]
+impl Retrieve<CustomStrategy<query::VectorQuery>> for LanceDB {
```

Comment:
Lancedb has a (not judging) bit confusing setup with concrete types starting at Query, which can be converted to a VectorQuery, they implement QueryBase and ExecutableQuery. I think ExecutableQuery is probably the type you want, but it's not dyn compatible (might still work with generics). Then a single implementation would work for both query types.